packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/TabPanels/PaymentTabPanel/PaymentRecordForm.tsx

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { components } from '@/api';
+import { RHFDatePicker } from '@/components/RHF/RHFDatePicker';
+import { RHFTextField } from '@/components/RHF/RHFTextField';
+import { useCurrencies } from '@/core/hooks';
+// import { RHFTimePicker } from '@/components/RHF/RHFTimePicker'; // Hidden for future iteration.
+import { yupResolver } from '@hookform/resolvers/yup';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import {
+  Button,
+  CircularProgress,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+  Grid,
+} from '@mui/material';
+
+import { PaymentRecordDetails } from './RecordManualPaymentModal';
+import { manualPaymentRecordValidationSchema } from './schemas/manualPaymentRecordValidationSchema';
+
+export type PaymentRecordFormValues = Omit<PaymentRecordDetails, 'created_by'>;
+type Props = {
+  invoice: components['schemas']['InvoiceResponsePayload'];
+  initialValues?: PaymentRecordFormValues;
+  isLoading: boolean;
+  onCancel: () => void;
+  onSubmit: (data: PaymentRecordFormValues) => void;
+};
+
+export const PaymentRecordForm: React.FC<Props> = ({
+  invoice,
+  initialValues,
+  isLoading,
+  onSubmit,
+  onCancel,
+}) => {
+  const { i18n } = useLingui();
+  const { formatCurrencyToDisplay } = useCurrencies();
+
+  const { control, handleSubmit, reset } = useForm<PaymentRecordFormValues>({
+    resolver: yupResolver(
+      manualPaymentRecordValidationSchema(i18n, invoice.amount_due)
+    ),
+    defaultValues: useMemo(
+      () =>
+        initialValues ?? {
+          amount: 0,
+          // reference_number: '', // Hidden for future iteration.
+          payment_date: null,
+          // payment_time: null, // Hidden for future iteration.
+        },
+      [initialValues]
+    ),
+  });
+
+  useEffect(() => {
+    reset(initialValues);
+  }, [initialValues, reset]);
+
+  const handlePaymentRecordSubmition = (
+    event: React.FormEvent<HTMLFormElement>
+  ) => {
+    event.preventDefault();
+
+    handleSubmit(onSubmit)(event);
+  };
+
+  return (
+    <>
+      <form onSubmit={handlePaymentRecordSubmition}>
```

Comment:
Could we avoid creating a separate function that only wraps another function? Defining handlePaymentRecordSubmition here simply creates an extra layer of indirection without adding any real value. Instead, it would be more efficient to pass handleSubmit(onSubmit) directly as the onSubmit handler in the form, reducing unnecessary wrapping

Reply:
I agree.